src/mod_ping.erl

```diff
@@ -152,30 +122,60 @@ iq_ping(Acc, _From, _To, #iq{sub_el = SubEl} = IQ, _) ->
 %% Hook callbacks
 %%====================================================================
 
-handle_remote_hook(HandlerState, mod_ping, Args, C2SState) ->
-    handle_remote_call(Args,
-                       ejabberd_c2s_state:jid(C2SState),
-                       ejabberd_c2s_state:server(C2SState),
-                       ejabberd_c2s_state:host_type(C2SState),
-                       HandlerState);
-handle_remote_hook(HandlerState, _, _, _) ->
-    HandlerState.
-
-user_online(Acc, _HostType, {_, Pid} = _SID, _Jid, _Info) ->
-    ejabberd_c2s:run_remote_hook(Pid, mod_ping, init),
-    Acc.
-
-user_offline(Acc, {_, Pid} = _SID, _JID, _Info, _Reason) ->
-    ejabberd_c2s:run_remote_hook(Pid, mod_ping, remove_timer),
-    Acc.
-
-user_send(Acc, _JID, _From, _Packet) ->
-    ejabberd_c2s:run_remote_hook(self(), mod_ping, init),
-    Acc.
-
-user_keep_alive(Acc, _JID) ->
-    ejabberd_c2s:run_remote_hook(self(), mod_ping, init),
-    Acc.
+-spec user_send_iq(mongoose_acc:t(), mongoose_c2s:hook_params(), map()) ->
+    gen_hook:hook_fn_ret(mongoose_acc:t()).
+user_send_iq(Acc, #{c2s_data := StateData}, #{host_type := HostType}) ->
+    case {mongoose_acc:stanza_type(Acc),
+          mongoose_c2s:get_mod_state(StateData, ?MODULE)} of
+        {<<"result">>, #ping_handler{id = PingId, time = T0}} ->
+            IqResponse = mongoose_acc:element(Acc),
+            IqId = exml_query:attr(IqResponse, <<"id">>),
+            case {IqId, PingId} of
+                {Id, Id} ->
+                    Jid = mongoose_c2s:get_jid(StateData),
+                    TDelta = erlang:monotonic_time(millisecond) - T0,
+                    mongoose_hooks:user_ping_response(HostType, #{}, Jid, IqResponse, TDelta),
+                    Action = {{timeout, ping_timeout}, cancel},
+                    {stop, mongoose_c2s_acc:to_acc(Acc, actions, Action)};
+                _ ->
+                    {ok, Acc}
+            end;
+        {_, {error, not_found}} ->
```

Comment:
This clause is redundant as a next one returns the same result.

Reply:
Indeed 👌🏽